src/components/AvatarWithIndicator.js

```diff
@@ -14,37 +19,116 @@ const propTypes = {
-};
+class AvatarWithIndicator extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.rotate = new Animated.Value(0);
+        this.scale = new Animated.Value(1);
+        this.startSyncIndicator = this.startSyncIndicator.bind(this);
+        this.stopSyncIndicator = this.stopSyncIndicator.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.isSyncing) {
+            this.startSyncIndicator();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.isSyncing && this.props.isSyncing) {
+            this.startSyncIndicator();
+        } else if (prevProps.isSyncing && !this.props.isSyncing) {
+            this.stopSyncIndicator();
+        }
+    }
+
+    componentWillUnmount() {
+        this.stopSyncIndicator();
+    }
+
+    startSyncIndicator() {
+        Animated.loop(Animated.timing(this.rotate, {
+            toValue: 1,
+            duration: 3000,
+            easing: Easing.linear,
+            isInteraction: false,
+            useNativeDriver: true,
+        })).start();
+        Animated.spring(this.scale, {
+            toValue: 1.35,
+            tension: 1,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start();
+    }
+
+    stopSyncIndicator() {
+        Animated.spring(this.scale, {
+            toValue: 1,
+            tension: 1,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start(() => {
+            this.rotate.resetAnimation();
+            this.scale.resetAnimation();
+            this.rotate.setValue(0);
+        });
+    }
+
+    render() {
+        const indicatorStyles = [
+            styles.alignItemsCenter,
+            styles.justifyContentCenter,
+            {
+                transform: [{
+                    rotate: this.rotate.interpolate({
+                        inputRange: [0, 1],
+                        outputRange: ['0deg', '-360deg'],
+                    }),
+                }, {
+                    scale: this.scale,
+                }],
+            },
+            this.props.isSyncing ? styles.statusIndicatorSyncing : null,
+            this.props.size === 'large' ? styles.statusIndicatorLarge : styles.statusIndicator,
+            this.props.isActive ? styles.statusIndicatorOnline : styles.statusIndicatorOffline,
+        ];
+
+        return (
+            <View
+                style={[this.props.size === 'large' ? styles.avatarLarge : styles.sidebarAvatar]}
+            >
+                <Avatar
+                    style={[this.props.size === 'large' ? styles.avatarLarge : null]}
+                    source={this.props.source}
+                />
+                <Animated.View style={StyleSheet.flatten(indicatorStyles)}>
+                    {this.props.isSyncing && (
+                        <Icon
+                            src={Sync}
+                            fill={themeColors.textReversed}
+                            width="100%"
```

Comment:
Could you try making the icon 12x12? This way there is a little bit of margin around the icon inside of the 16x16 green circle. 

![image](https://user-images.githubusercontent.com/2319350/117890858-c49f8c00-b26a-11eb-809a-a450d5cca1b3.png)

Reply:
@shawnborton But indicator size is 12x12 including 2px transparent border around. So Green circle is 8x8.

Reply:
We want the indicator to become larger when we are showing that content is syncing. So the green online indicator should grow in size and then show the spinning icon inside of it. Let me know if that makes sense - happy to make a prototype as well.

Reply:
OK after growing, it should be 16px. makes sense.

Reply:
@shawnborton It looks a little big. Don't you think so?
![image](https://user-images.githubusercontent.com/24370807/118180824-a1d0cb80-b454-11eb-9beb-6cf541a87231.png)

Reply:
Or this is looking fine, let me know I will update the PR.

Reply:
@shawnborton any thoughts?

Reply:
I agree it looks big. I think ideally the size of the indicator wouldn't need to change – we just show the spinning icon in the normal-size indicator if it's loading.

Reply:
We can try a smaller size, which would be a green circle of 12x12, with the icon inside of it at 8x8. It would look like this:
![image](https://user-images.githubusercontent.com/2319350/118307600-c4d49d00-b49f-11eb-850c-22a198ba6696.png)

![image](https://user-images.githubusercontent.com/2319350/118307632-ce5e0500-b49f-11eb-96c0-1d2fccda317c.png)